core/src/smartcontracts/isi/account.rs

```diff
@@ -15,6 +15,59 @@ use crate::{ValidQuery, WorldStateView};
 pub mod isi {
     use super::{super::prelude::*, *};
 
+    #[allow(clippy::panic_in_result_fn)]
+    impl<W: WorldTrait> Execute<W> for Register<Asset> {
+        type Error = Error;
+
+        #[metrics(+"register_asset")]
+        fn execute(
+            self,
+            _authority: <Account as Identifiable>::Id,
+            wsv: &WorldStateView<W>,
+        ) -> Result<(), Self::Error> {
+            let asset_id = self.object.id().clone();
+
+            match wsv.asset(&asset_id) {
+                Err(e) if matches!(e, FindError::Asset(_)) => {
+                    let res = wsv.modify_account(&asset_id.account_id, |account| {
+                        // We are in an arm where asset doesn't exist
+                        assert!(account.add_asset(self.object).is_none());
```

Comment:
What's the difference with `unwrap()`? Style? 🤔

Reply:
It seems to me that it's more appropriate semantically, since it emphasizes that this should be unreachable state, not just unrecoverable. `unreachable!` would be even better semantically, but it would look awkward.